apps/betterangels-backend/clients/types.py

```diff
@@ -104,10 +104,34 @@ def search(
 
         queryset = queryset.filter(reduce(and_, q_objects))
 
-        return (
-            queryset,
-            Q(),
-        )
+        return (queryset, Q())
+
+    @strawberry_django.filter_field
+    def full_name_search(
```

Comment:
Is there a reason we couldn't use or update `search` above for functionality? 
Unless there's a strawberry related (some other ) reason, would we want to have a single general purpose search function for this model?

Reply:
yeah, just noticed that. seems like a very specific implementation of `search`, but it happens

Reply:
Typically I’d imagine the interface to a generic search function to accept named params, so if I wanted to search for a client , and passed in `first_name + last_name` then it would search by those 2 fields. 
If I passed in `first_name + nickname + last_name` it would search by those 3 fields, etc…. 

Otherwise we’ll need to write a separate search function for each type of search.
A generic function, in theory, could also accept options for things such as:
* case sensitivity
* exact or partial match
* etc…

That’s not requested here, but I'd hope we could come back and extend this function to add that functionality.
If there are other limitations here then ok

Having said that, I can also defer to @mikefeldberg and Paul